### PR TITLE
Added the 'GetExpectedParameterTypes' method to the PreparedStatement…

### PR DESCRIPTION
**Pull request by AlexR2D2**
This method allows to get the expected types of parameters of SQL query.

To be short.
DuckDB amalgamation can be used in the Erlang NIF. In the Erlang the parameters of SQL query are passed as terms. [Term](https://www.erlang.org/doc/reference_manual/data_types.html) - a piece of data of any data type. We can use special Erlang NIF functions like enif_is_number(term), enif_is_binary(term) to detect the type of the term we passed into query and convert this term parameter to the duckdb::Value parameter of this detected type. 

Basically, it works, but may be it would be more correct if we could get the expected types of parsed SQL query parameters from the duckdb::PreparedStatement not from the terms itself, and try convert Erlang terms to the duckdb::Value's of expected types.

### CODE REVIEW

**Comment by Tishj** [comment]
Can we add a `D_ASSERT(it.first >= 1);`?
To cover the assumption made when doing `it.first - 1`

**Comment by AlexR2D2** [comment]
Sure, updated. If you want, you can make any changes

**Comment by hannes** [comment]
We might want to think about how this could be extended to named parameters.

**Comment by AlexR2D2** [comment]
> We might want to think about how this could be extended to named parameters.

Maybe then replace this:
vector<LogicalType> PreparedStatement::GetExpectedParameterTypes()
to this?
unordered_map<string, LogicalType> PreparedStatement::GetExpectedParameterTypes()

For example, for SQL parameters like this _$1_, _$2_, _$name_, _$city_ it will returns unordered_map like this _{"1" -> LogicalType, "2" -> LogicalType, "name" -> LogicalType, "city" -> LogicalType}_

**Comment by Mytherin** [comment]
I would say the named parameters should be separate from the positional parameters - this looks fine to me for the positional parameters.
